src/hardware/memory.cpp

```diff
@@ -564,9 +564,6 @@ class MEMORY:public Module_base{
 			LOG_MSG("Stick with the default values unless you are absolutely certain.");
 		}
 		MemBase = new Bit8u[memsize*1024*1024];
-		if (!MemBase) E_Exit("Can't allocate main memory of %d MB",memsize);
```

Comment:
(just a comment)
If we would like to preserve `E_Exit` with a helpful message in here, then line 566 could be rewritten as:

    MemBase = new (std::nothrow) uint8_t[memsize * 1024 * 1024];

Reply:
Great suggestion: if we're going to hit allocation issues, it'll be here! 
Also added a couple other messages to adjacent (but surely smaller) allocations.